src/SigasiProjectCreator/DotF/DotFfileParser.py

```diff
@@ -9,12 +9,25 @@
 import os
 import sys
 import glob
+import re
 
 from SigasiProjectCreator.ArgsAndFileParser import ArgsAndFileParser
 from .parseFile import parse_dotf
 from ..convertDotFtoCsv import rebase_file
 
 
+def abspath(path):
+    s_path = str(path)
+    if s_path.startswith('\\') or s_path.startswith('/') or s_path[1] == ':' or s_path.startswith('$'):
+        # this is an absolute path in Linux or Windows
+        return path
+    return os.path.abspath(path)
+
+
+def expandvars_plus(s):
+    return os.path.expandvars(re.sub(r'\$\((.*)\)', r'${\1}', s))
```

Comment:
So... `s` may contain parts marked `$(stuff)`, you then transform those into `${stuff}` and then run shell variable expansion? Please document this and why we need this. It seems to me the user should just use `${...}` and let it be replaced by their shell.

Reply:
The `$(...)` syntax occurs in the RiscV repo which we're trying to support.